qiskit/synthesis/__init__.py

```diff
@@ -89,6 +89,28 @@
 
 .. autofunction:: synth_qft_line
 
+Unitary Synthesis
+=================
```

Comment:
It might be nice to add brief summaries of these submodules here and highlight why 1q and 2q synthesis is not inside the `Unitary Synthesis` heading 🙂